app/code/Magento/UrlRewrite/Model/StoreSwitcher/RewriteUrl.php

```diff
@@ -67,13 +67,18 @@ public function switch(StoreInterface $fromStore, StoreInterface $targetStore, s
         ]);
         if ($oldRewrite) {
             // look for url rewrite match on the target store
-            $currentRewrite = $this->urlFinder->findOneByData([
-                UrlRewrite::REQUEST_PATH => $urlPath,
+            $currentRewrites = $this->urlFinder->findAllByData([
+                UrlRewrite::TARGET_PATH => $oldRewrite->getTargetPath(),
                 UrlRewrite::STORE_ID => $targetStore->getId(),
             ]);
-            if (null === $currentRewrite) {
+            if (null === $currentRewrites || empty($currentRewrites)) {
                 /** @var \Magento\Framework\App\Response\Http $response */
                 $targetUrl = $targetStore->getBaseUrl();
+                return $targetUrl;
+            }
+            foreach ($currentRewrites as $rewrite) {
+                $targetUrl = $targetStore->getBaseUrl() . $rewrite->getRequestPath();
+                break;
```

Comment:
maybe it's just personal preference, but returning the targetUrl here instead of a break would make it consistent with the if condition before and so maybe make it more readable.

Reply:
Hi @davidverholen 
I have updated code as per your instruction, there were no need to break statement, I have removed it. I have checked URL rewrite functionality with multiple store in my local setup, it is working fine. 

Thanks.